src/formatDate.js

```diff
@@ -8,7 +8,35 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separatorFrom = fromFormat.pop();
+  const separatorTo = toFormat.pop();
+  const dateArr = date.split(separatorFrom);
+  const resultArr = [];
+
+  for (let i = 0; i < toFormat.length; i++) {
+    const format = toFormat[i];
+    const index = fromFormat.indexOf(format);
+
+    if (index === -1) {
+      if (format === 'YYYY') {
+        const fromIndex = fromFormat.indexOf('YY');
+        const year = dateArr[fromIndex];
+
+        resultArr.push(year < 30 ? '20' + year : '19' + year);
+      } else if (format === 'YY') {
+        const fromIndex = fromFormat.indexOf('YYYY');
+        const year = dateArr[fromIndex];
+
+        resultArr.push(year.slice(-2));
+      }
+    } else {
+      resultArr.push(dateArr[index]);
+    }
```

Comment:
The logic for converting `YY` to `YYYY` and vice versa does not account for the case where `YY` or `YYYY` is not found in `fromFormat`. This will cause an issue when `fromIndex` is `-1` and you try to access `dateArr[fromIndex]`. You need to handle the case where the year format is not present in the `fromFormat` array.